query/physicalplan/binaryscalarexpr.go

```diff
@@ -201,9 +100,52 @@ func BinaryScalarOperation(left arrow.Array, right scalar.Scalar, operator logic
 		panic("TODO: list comparisons unimplemented")
 	}
 
+	switch operator {
+	case logicalplan.OpEq:
+		return ArrayScalarCompute("equal", left, right)
```

Comment:
Are these consts anywhere in the compute pkg?

Reply:
nope, they are hard coded in each kernel when functions are being dynamically registered. We can do better by resolving to these kernel function names `i.e` `(logicalplan.OpEq).String() =="equal"` so we can have constants of our own,  completely  removing those switch case.

Reply:
I was looking for those constants. You're right that we don't even need those. 

Wow. 😄